bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/view/obs/ObsListView.scala

```diff
@@ -339,42 +339,45 @@ class ObsListView(shellAdvisor:ShellAdvisor, band:Band, queueLookup: Target => U
       }
     }
 
-    // Our double-click handler for editing
-    onDoubleClick {
-
-      // Edit handlers for groups
-      case cg:ConditionGroup => edit(cg, p => ConditionEditor.open(cg.c, canEdit, panel), Observation.condition)
-      case bg:BlueprintGroup => edit(bg, BlueprintEditor.open(_, bg.b, canEdit, panel), Observation.blueprint)
-      case tg:TargetGroup    =>
+    // Method for handling the editing of the target group, since this is quite unique.
+    def editTargetGroup(tg: TargetGroup): Unit = {
```

Comment:
While this looks deceptively complicated, it is just a slight refactoring. I thought shoving the code for `case tg:TargetGroup` into this was messy, so I factored it out into its own method, `editTargetGroup`.